include/Session.h

```diff
@@ -33,8 +33,11 @@ class Session{
 
     std::vector<std::string> extractTags(nlohmann::json& tagList);
     void fillContentFromJson(const std::string &configFilePath);
-    template<T>
+    template<typename T>
     void cleanIterable(T* toDelete);
     void clean();
+    template <typename T>
+    void deepCopyPointerVector(const std::vector<T>& newV, std::vector<T>& ourV);
```

Comment:
why do you need to pass ourV? use "this"

and better convention is calling the variable "other"

Reply:
Because I want the same function for content and for actionsLog but the function is called in Session, which means "this" would be of type `Session*`